splitio/engine/impressions.py

```diff
@@ -1,160 +1,19 @@
 """Split evaluator module."""
-import threading
```

Comment:
i'd still advise to use a nested package approach tu structure all this. The splitio/engine package is not meant for impressions only. it'd be ideal to have a sub-package impressions inside it ant the rest of the impressions-related modules inside it as follows:
```
splitio/engine/impressions/__init__.py
splitio/engine/impressions/adapters.py
splitio/engine/impressions/manager.py
splitio/engine/impressions/strategies.py
splitio/engine/impressions/unique_keys_tracker.py
```

the __init__.py could even contain a method that does most of the configuration, to prevent cluttering the factory.py with that many if/else branches according to the impression mode